tools/cider-data-converter/src/converter.rs

```diff
@@ -165,66 +164,75 @@ fn unroll_float(
     val: f64,
     format: &interp::serialization::FormatInfo,
     round_float: bool,
-) -> impl Iterator<Item = u8> {
-    if let &interp::serialization::FormatInfo::Fixed {
-        signed,
-        int_width,
-        frac_width,
-    } = format
-    {
-        let rational = float_to_rational(val);
-
-        let frac_part = rational.fract().abs();
-        let frac_log = log2_exact(&frac_part.denom().to_biguint().unwrap());
-
-        let number = if frac_log.is_none() && round_float {
-            let w = BigInt::from(1) << frac_width;
-            let new = (val * w.to_f64().unwrap()).round();
-            new.to_bigint().unwrap()
-        } else if frac_log.is_none() {
-            panic!("Number {val} cannot be represented as a fixed-point number. If you want to approximate the number, set the `round_float` flag to true.");
-        } else {
-            let int_part = rational.to_integer();
-
-            let frac_log = frac_log.unwrap_or_else(|| panic!("unable to round the given value to a value representable with {frac_width} fractional bits"));
-            if frac_log > frac_width {
-                panic!("cannot represent value with {frac_width} fractional bits, requires at least {frac_log} bits");
-            }
+) -> Vec<u8> {
```

Comment:
What's the reasoning for forcing this to be a `Vec` over an iterator?

Reply:
Because you cannot return different types of iterators. Since the existing branch returns a `Take<...>` iterator, I would have to return the exact same type from the new branch. The only alternative would be to box it.

Reply:
ohhhhh, right duh! Missed the fact that the concrete types were different. I think just returning the vec makes more sense than boxing a `dyn` Iterator to avoid the allocations